x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/AutodetectBuilder.java

```diff
@@ -365,6 +372,28 @@ private void buildFieldConfig(List<String> command) throws IOException {
         }
     }
 
+    private void buildScheduledEventsConfig(List<String> command) throws IOException {
+        if (scheduledEvents.isEmpty()) {
+            return;
+        }
+        Path eventsConfigFile = Files.createTempFile(env.tmpFile(), "eventsConfig", JSON_EXTENSION);
+        filesToDelete.add(eventsConfigFile);
+
+        List<DetectionRuleWriter> detectionRuleWriters = scheduledEvents.stream()
```

Comment:
As we're streaming scheduled events, I think it would be clearer to map these to a `ScheduledEventToRuleWriter`. Then the `RULES` `ParseField` added in `ScheduledEvent` can move in that writer class.